main/guides/zoe/contract-basics.md

```diff
@@ -1,46 +1,46 @@
 
-Let's start with a contract with a simple `greet` function:
+For the hello-world smart contract, we will have a simple `greet` function apart from `start` function. The `greet` function takes a string as a parameter (for example, name of the person calling the function) and returns a customized greeting message.
 
 <<< @/../snippets/zoe/src/01-hello.js#greet
 
-The `start` function can expose the `greet` function
-as part of the contract API by making it
-a method of the contract's `publicFacet`:
+The `greet` function, along with any other public function, must be made accessible through the `publicFacet` of the contract. The `start` function returns an object with a `publicFacet` property. In the hello-world contract, the `start` function exposes the `greet` function by defining it as a method of the contract's `publicFacet`, as shown below:
 
 <<< @/../snippets/zoe/src/01-hello.js#publicFacet
 
-We mark it `Far(...)` to allow callers to use it from outside the contract
-and give it a suggestive interface name for debugging.
+We wrap the value of the `publicFacet` property in a `Far(...)` call to safely expose it as a remote object, accessible from outside the contract. This also gives it a suggestive interface name `Hello` for debugging.
 _We'll discuss [Far in more detail](../js-programming/far) later._
 
 Putting it all together:
 
 <<< @/../snippets/zoe/src/01-hello.js#contract
 
+Let us save this code to a file named `01-hello.js` inside `src` directory. 
 ## Using, testing a contract
 
-Let's use some tests to explore how a contract is used.
+Agoric contracts are typically tested using the [ava](https://github.com/avajs/ava) framework. The test file begins with an `import @endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment. We also import `E()` in order to make asynchronous method calls and `test` function from `ava`. _We'll talk more about [using `E()` for async method calls](../js-programming/eventual-send) later._ Following these `import` statements, we write a simple test that validates that the `greet` method works as expected.
 
-Agoric contracts are typically tested
-using the [ava](https://github.com/avajs/ava) framework.
-They start with `@endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment:
-
-<<< @/../snippets/zoe/contracts/test-zoe-hello.js#test-imports
+Putting it all together:
 
-_We'll talk more about [using `E()` for async method calls](../js-programming/eventual-send) later._
+<<< @/../snippets/zoe/contracts/test-zoe-01-hello.js#test-01-hello
 
-A test that the `greet` method works as expected looks like:
+Let's save this code in a file named `test-01-hello.js` in a `test` directory. Both `src` and `test` directories should lie in the same `contract` directory. Let us run the following command to execute the test:
 
-<<< @/../snippets/zoe/contracts/test-zoe-hello.js#test1
+```sh
+ava --match="contract greets by name"
```

Comment:
oops! when I requested taking out `npx`, I should have offered a replacement:

```suggestion
yarn ava --match="contract greets by name"
```

I don't expect that invoking `ava` on its own will work unless somebody has done `npm install -g ava`.

not critical, I suppose

Reply:
Thanks @dckc. I am adding "follow the instructions there to install ava" right next to where we provide link for "ava". Probably we will fix this with something better soon.